sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/ArithmeticExpressionSuite.scala

```diff
@@ -376,4 +404,100 @@ class ArithmeticExpressionSuite extends SparkFunSuite with ExpressionEvalHelper
     Greatest(Seq(Literal(1), Literal(1))).genCode(ctx2)
     assert(ctx2.inlinedMutableStates.size == 1)
   }
+
+  test("SPARK-24598: overflow on long returns wrong result") {
+    val maxLongLiteral = Literal(Long.MaxValue)
+    val minLongLiteral = Literal(Long.MinValue)
+    val e1 = Add(maxLongLiteral, Literal(1L))
+    val e2 = Subtract(maxLongLiteral, Literal(-1L))
+    val e3 = Multiply(maxLongLiteral, Literal(2L))
+    val e4 = Add(minLongLiteral, minLongLiteral)
+    val e5 = Subtract(minLongLiteral, maxLongLiteral)
+    val e6 = Multiply(minLongLiteral, minLongLiteral)
```

Comment:
shall we also test `UnaryMinus`? BTW do you know why the previous mistake in `UnaryMinus` code was not caught by the existing tests?

Reply:
yes, it was my fault, I was not testing "normal" cases with the flag turned on. I fixed it. `UnaryMinus` cases are already tested in its UT, I think it is useless to add them here too.

Reply:
makes sense, thanks!